spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/ReactorLoadBalancerClient.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.loadbalancer.reactive;
+
+import java.net.URI;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * Represents a <a href="https://projectreactor.io/">Project-Reactor</a>-based client-side
+ * load balancer.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 2.2.0
+ */
+
+public interface ReactorLoadBalancerClient {
```

Comment:
I'm confused about this new class when we already have `ReactiveLoadBalancer`.

Reply:
that's the client that will retrieve an instance of the `ReactiveLoadBalancer` to use underneath based on the `serviceId` (will load either from default configuration or provided by `@LoadBalancerClient`. It also handles reconstructing the URI that is later used by the `ReactorLoadBalancerExchangeFunction`, but I can move this logic elsewhere. Please let me know where you think I should move it to.

Reply:
Changed.